transforms/ember-object/index.js

```diff
@@ -0,0 +1,197 @@
+  const j = api.jscodeshift;
+  const root = j(file.source);
+
+  // Helper functions
+  /**
+   * Transform instance properties to MemberExpressions
+   *
+   * For example: `prop: value` --> `this.prop = value`
+   *
+   * @param {*} instanceProps
+   * @returns Array [ ExpressionStatement ]
+   */
+  function instancePropsToExpressions(instanceProps) {
+    return instanceProps.map(instanceProp =>
+      j.expressionStatement(
+        j.assignmentExpression(
+          "=",
+          j.memberExpression(j.thisExpression(), instanceProp.key),
+          instanceProp.value
+        )
+      )
+    );
+  }
+
+  /**
+   * Transform functions to class methods
+   *
+   * For example { foo: function() { }} --> { foo() { }}
+   *
+   * @param {[]} functionProps
+   * @returns Array [MethodDefinition]
+   */
+  function functionPropsToMethods(functionProps = []) {
+    return functionProps.map(functionProp => {
+      return j.methodDefinition("method", functionProp.key, functionProp.value);
+    });
+  }
+
+  /**
+   * Create  a constructor method
+   *
+   * @param {[]} instanceProps
+   * @return Array [MethodDefinition]
+   */
+  function createConstructor(instanceProps = []) {
+    if (instanceProps.length) {
+      return [
+        j.methodDefinition(
+          "constructor",
+          j.identifier("constructor"),
+          j.functionExpression(
+            null,
+            [],
+            j.blockStatement(instancePropsToExpressions(instanceProps))
+          )
+        )
+      ]; // Todo: find a way to include params
+    }
+
+    return [];
+  }
+
+  /**
+   * Create the class
+   *
+   * @param {String} className
+   * @param {Object} { instanceProps, functionProps } map of instance and function properties
+   */
+  function createClass(className, { instanceProps, functionProps } = {}) {
+    return j.classDeclaration(
+      j.identifier(className),
+      j.classBody(
+        []
+          .concat(createConstructor(instanceProps))
+          .concat(functionPropsToMethods(functionProps))
+      ),
+      j.identifier("EmberObject")
+    );
+  }
+
+  /**
+   * Return the map of instance props and functions from Ember Object
+   *
+   * For example
+   * const myObj = EmberObject.extend({ key: value, foo: function() {}, baz() {} });
+   * will be parsed as:
+   *{
+      instanceProps: [ key: value ]
+      functionProps: [ FunctionExpression(foo), FunctionExpression(baz) ]
+    }
+   * @param {*} emberObjectExpression
+   * @returns { instanceProps, functionProps } map of instance and function properties
+   */
+  function getEmberObjectProperties(emberObjectExpression) {
+    const objProps = j(emberObjectExpression)
+      .find(j.ObjectExpression)
+      .get("properties").value;
+
+    const instanceProps = [];
+    const functionProps = [];
+
+    objProps.forEach(prop => {
+      if (prop.value.type === "FunctionExpression") {
+        functionProps.push(prop);
+      } else {
+        instanceProps.push(prop);
```

Comment:
This can probably wait until later, but we'll need to add checks for computed properties and other non-standard instance props here